music_assistant/providers/audible/audible_helper.py

```diff
@@ -76,85 +107,158 @@ async def get_library(self) -> AsyncGenerator[Audiobook, None]:
             )
 
             items = library.get("items", [])
+            total_items = library.get("total_results", 0)
+
+            self.logger.debug(
+                "Audible: Got %s items (total reported by API: %s)", len(items), total_items
+            )
+
+            # Check if this page had no items (which means we've reached the end)
             if not items:
+                self.logger.debug(
+                    "Audible: No more items returned, ending pagination (processed %s items)",
+                    total_processed,
+                )
                 break
 
+            # Process items in this page
+            items_processed_this_page = 0
             for audiobook_data in items:
+                # Check if this is a podcast or non-audio asset that we should skip
+                content_type = audiobook_data.get("content_delivery_type", "")
+                if content_type in ("PodcastParent", "NonAudio"):
+                    self.logger.debug(
+                        "Skipping non-audiobook item: %s (%s)",
+                        audiobook_data.get("title", "Unknown"),
+                        content_type,
+                    )
+                    total_processed += 1
+                    continue
+
                 asin = audiobook_data.get("asin")
                 cached_book = await self.mass.cache.get(
                     key=asin, base_key=CACHE_DOMAIN, category=CACHE_CATEGORY_AUDIOBOOK, default=None
                 )
 
-                if cached_book is not None:
-                    album = await self._parse_audiobook(cached_book)
-                    yield album
-                else:
-                    album = await self._parse_audiobook(audiobook_data)
-                    yield album
-
-            # Check if we've reached the end
-            total_items = library.get("total_results", 0)
-            if page * page_size >= total_items:
-                break
+                try:
+                    if cached_book is not None:
+                        album = await self._parse_audiobook(cached_book)
+                        yield album
+                    else:
+                        album = await self._parse_audiobook(audiobook_data)
+                        yield album
+
+                    total_processed += 1
+                    items_processed_this_page += 1
+                except ValueError as exc:
+                    self.logger.warning(f"Skipping invalid audiobook: {exc}")
+                    total_processed += 1
+                    continue
+
+            self.logger.debug(
+                "Audible: Processed %s valid audiobooks on page %s", items_processed_this_page, page
+            )
 
+            # Continue to the next page
             page += 1
+            self.logger.debug(
+                "Audible: Moving to page %s (processed: %s, total reported: %s)",
+                page,
+                total_processed,
+                total_items,
+            )
+
```

Comment:
maybe also break out the loop if the number of items is less than the page size ?